.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,64 @@
+name: "CodeQL"
+
+on:
+    push:
+    workflow_dispatch:
+    pull_request:
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python' ]
+        # Learn more:
+        # https://docs.github.com/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#changing-the-languages-that-are-analyzed
+
+    steps:
+    - uses: actions/checkout@v4
+      with:
+       submodules: recursive
+
+    - name: Generate (dummy) demo_config.h
+      run: |
+             cp examples/demo_config/demo_config_template.h examples/demo_config/demo_config.h
+             sed -i '/^#if defined( AWS_ACCESS_KEY_ID ) && defined( AWS_IOT_THING_ROLE_ALIAS )/i\
+             #define AWS_CREDENTIALS_ENDPOINT ""\
+             #define AWS_IOT_THING_NAME ""\
+             #define AWS_IOT_THING_ROLE_ALIAS ""\
+             #define AWS_IOT_THING_CERT_PATH "certificate.pem"\
+             #define AWS_IOT_THING_PRIVATE_KEY_PATH "private.key"\
+             ' examples/demo_config/demo_config.h
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v3
+      with:
+        languages: ${{ matrix.language }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
+        # By default, queries listed here will override any specified in a config file.
+        # Prefix the list here with "+" to use these queries and those in the config file.
+        # queries: ./path/to/local/query, your-org/your-repo/queries@main
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v3
+
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
+    #- run: |
+    #   make bootstrap
+    #   make release
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v3
```

Comment:
```suggestion
      uses: github/codeql-action/analyze@v3

```
Nit: append an empty line at the end of file.

Reply:
thank you - we also need a format checker github action ;)